source/agora/consensus/protocol/Nominator.d

```diff
@@ -526,15 +526,36 @@ extern(D):
         }
 
         const timer_id = ++this.last_timer_id[type];
+
+        // this.taskman.runTask(
+        // {
+        //     this.taskman.wait(timeout.msecs);
+
+        //     // timer was cancelled
+        //     if (timer_id < this.active_timer_ids[type])
+        //         return;
+
+        //     callCPPDelegate(callback);
+        // });
+
         this.taskman.runTask(
         {
-            this.taskman.wait(timeout.msecs);
+            ITimer timer;
 
-            // timer was cancelled
-            if (timer_id < this.active_timer_ids[type])
-                return;
+            void callDelegate ()
+            {
+                // timer was cancelled
+                if (timer_id < this.active_timer_ids[type])
+                {
+                    timer.stop();
+                    return;
+                }
+                callCPPDelegate(callback);
+            }
 
-            callCPPDelegate(callback);
+            callDelegate(); // avoid delay
```

Comment:
This is also wrong. Notice that there was a `wait` and then it was called. There is a specific reason for this. These are timeout routines.

Reply:
I'll show you exactly how to do it.